addons/binding/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/handler/FridgeFreezerChannelSelector.java

```diff
@@ -0,0 +1,169 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.miele.handler;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map.Entry;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.miele.handler.MieleBridgeHandler.DeviceMetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+
+/**
+ * The {@link ApplianceChannelSelector} for fridges with
+ * a freezer compartment
+ *
+ * @author Karel Goderis - Initial contribution
+ */
+public enum FridgeFreezerChannelSelector implements ApplianceChannelSelector {
+
+    PRODUCT_TYPE("productTypeId", "productType", StringType.class, true),
+    DEVICE_TYPE("mieleDeviceType", "deviceType", StringType.class, true),
+    BRAND_ID("brandId", "brandId", StringType.class, true),
+    COMPANY_ID("companyId", "companyId", StringType.class, true),
+    STATE("state", "state", StringType.class, false),
+    FREEZERSTATE("freezerState", "freezerstate", StringType.class, false),
+    FRIDGESTATE("fridgeState", "fridgestate", StringType.class, false),
+    SUPERCOOL(null, "supercool", OnOffType.class, false),
+    SUPERFREEZE(null, "superfreeze", OnOffType.class, false),
+    FREEZERCURRENTTEMP("freezerCurrentTemperature", "freezercurrent", DecimalType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            return getState(s);
+        }
+    },
+    FREEZERTARGETTEMP("freezerTargetTemperature", "freezertarget", DecimalType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            return getState(s);
+        }
+    },
+    FRIDGECURRENTTEMP("fridgeCurrentTemperature", "fridgecurrent", DecimalType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            return getState(s);
+        }
+    },
+    FRIDGETARGETTEMP("fridgeTargetTemperature", "fridgetarget", DecimalType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            return getState(s);
+        }
+    },
+    DOOR("signalDoor", "door", OpenClosedType.class, false) {
+        @Override
+
+        public State getState(String s, DeviceMetaData dmd) {
+            if (s.equals("true")) {
+                return getState("OPEN");
```

Comment:
Maybe, but since I had some more string to string conversions elsewhere, I just sticked to string I think. You are right, but I wanted to be homogeneous in the source